### PR TITLE
Hardsuit RAD and EMP shield module implementations.

### PR DESCRIPTION
**Pull request by DrSnips**
[balance] [content] [tweak]
A while back in #26951 a few new hardsuit modules were added but not implemented due to atomic and balance reasons. This is the (late) follow up PR to that one.

**From before:**
To restate what the modules do.
- The EMP dissipation module will consume your suit's charge(300/150/100 based on EMP severity) to prevent it from getting disabled/its battery damaged.
- The R.A.D. will shield your suit and helmet until it's filter is full (It will fill based on the amount of radiation and the suit's initial radiation shielding). You can use a SSU to clean the filter.
The RAD has a normal and advanced form each holding 250 and 800 rads respectively. To put it in perspective a suit with 10 rad armor to start will absorb about 1 blast of radiation if you were right next to a shard. A suit with 50 rad armor to start will absorb about 5-6 item blasts worth. The advanced one is about 3x that so you do the math. 
**TLDR** suits with better rad armor to start can take on more rads with the module. 

For people who are wondering if dragging the shard behind them while spraying it in the hall with this will be a cool idea:
<h3>You get two sprays and then your module gives in.</h3>
</br>

**Balance stuff:**
- Engineer and CE hard suits: -30 (80 -> 50) rad armor, +round start basic RAD module. This rad armor is in line with existing "good" rad armor amounts where they used to be the "best" rad armor on normally available hard suits.
- EMP module costs gold and the usual circuit materials. 
- RAD module costs plasma and usual circuit materials. 
- Advanced RAD module costs plasma, gold, and usual circuit materials. 
- All require mid level research and the RAD modules need plasma research.

**Why:**
When the shard is yelling 99% and you decide to take one for the team and enter the hot box to try and deal with the shard you are basically accepting your death sentence by internal bleeding because 80 rad armor means not much when standing next to an over stimmed shard. An actual rad suit means you die before you even get next to the shard from whatever is over stimming it in the first place. This makes engineering suits a happy medium between rad suits and pressure suits by being a rad suit for only a limited time.
Shard dragging and spraying in the halls is not a great idea since it's limited time (about 2 sprays worth). Atmos suits lack rad armor so they tank even less with the RAD.

**Coder stuff you won't care about:**
- Engineering hard suits are now a child of a basic hard suit under the name civilian suit. Nothing has changed stat wise besides the already mentioned changes and exist for cleaner tweaking of hard suits without having to touch the engineering hard suit always.
- Children of related parents have been updated where needed.
- I have updated references to engineering hard suits where appropriate in the code. Yell at me if you think I'm blind and I missed one.

:cl:
 * rscadd: EMP and RAD shield modules are now available in game by means similar to existing hard suit modules.
 * tweak: Engineer and CE hard suits have undergone these changes.
 * rscadd: Both are equipped with a basic RAD module at round start
 * rscdel: Both have lost some (80 -> 50) rad armor as balance. This amount is similar to most "good" rad armor hard suits.

### CODE REVIEW

**Comment by DrSnips** [comment]
>Conflicts
>With loadouts

the-simpsons-technical-difficulties-drunk

**Comment by Zth--** [comment]
the file you're having conflicts with was removed

File: code/modules/clothing/spacesuits/rig.dm [hunk 1]
```diff
@@ -8,7 +8,7 @@ var/list/all_hardsuit_pieces = list(HARDSUIT_HEADGEAR,HARDSUIT_GLOVES,HARDSUIT_B
 
 //Regular rig suits
 /obj/item/clothing/head/helmet/space/rig
```
Comment (Kurfursten) [inline code/modules/clothing/spacesuits/rig.dm:10]: seems weird to not update the armor stats on the "new" civilian models

our spriters would probably be willing to make you a generic grey rig too

Comment (DrSnips) [reply code/modules/clothing/spacesuits/rig.dm:10]: There's a technically grey one in the works currently but not by me fyi. I talked about it on the sprites channel in the discord.

File: code/modules/clothing/spacesuits/rig_modules/rig_modules.dm [hunk 2]
```diff
@@ -258,3 +258,8 @@
 
 	if(current_capacity >= max_capacity)
 		deactivate()
+
+/obj/item/rig_module/rad_shield/adv
+	name = "high capacity radiation absorption device"
+	desc = "Its acronym, R.A.D., and full name both convey the application of this module. By using similar technology as radiation collectors, it protects the suit wearer from incoming radiation until its collectors are full. It can be reset by using a suit storage unit's cleaning operation."
```
Comment (jknpj) [inline code/modules/clothing/spacesuits/rig_modules/rig_modules.dm]: duplicated desc?

Comment (DrSnips) [reply code/modules/clothing/spacesuits/rig_modules/rig_modules.dm]: Whoops I mist have lost the part of extended cap in the desc when doing rebase stuff. Will fix.

Comment (DrSnips) [reply code/modules/clothing/spacesuits/rig_modules/rig_modules.dm]: resolved
